catalog/app/utils/json-schema.js

```diff
@@ -0,0 +1,84 @@
+import * as R from 'ramda'
+import isArray from 'lodash/isArray'
+import isNumber from 'lodash/isNumber'
+import isObject from 'lodash/isObject'
+import isString from 'lodash/isString'
+import isUndefined from 'lodash/isUndefined'
+
+function isSchemaArray(optSchema) {
+  return R.prop('type', optSchema) === 'array'
+}
+
+function isSchemaObject(optSchema) {
+  return R.prop('type', optSchema) === 'object'
+}
+
+function isSchemaString(optSchema) {
+  return R.prop('type', optSchema) === 'string'
+}
+
+function isSchemaNumber(optSchema) {
+  return R.prop('type', optSchema) === 'number'
+}
+
+function isSchemaBoolean(optSchema) {
+  return R.prop('type', optSchema) === 'boolean'
+}
+
+function isSchemaNull(optSchema) {
+  return R.prop('type', optSchema) === 'null'
+}
+
+export function isSchemaEnum(optSchema) {
+  return Boolean(R.prop('enum', optSchema))
+}
+
+function isSchemaConst(optSchema) {
+  return Boolean(R.prop('const', optSchema))
+}
+
+function isSchemaCompound(optSchema) {
+  if (!optSchema) return false
+  return ['anyOf', 'oneOf', 'not', 'allOf'].some((key) => optSchema[key])
+}
+
+function isSchemaReference(optSchema) {
+  return Boolean(R.prop('$ref', optSchema))
+}
+
+export function isNestedType(optSchema) {
+  return isSchemaArray(optSchema) || isSchemaObject(optSchema)
+}
+
+export function schemaTypetoHumanString(optSchema) {
```

Comment:
should be "schemaType**To**HumanString" i believe.
also, this can be written in point-free style: `const schemaTypeToHumanString = R.cond([ ... ])`